pkg/statistics/handle/util/util.go

```diff
@@ -76,20 +76,23 @@ var (
 )
 
 // CallWithSCtx allocates a sctx from the pool and call the f().
-func CallWithSCtx(pool util.SessionPool, f func(sctx sessionctx.Context) error, flags ...int) (err error) {
+func CallWithSCtx(pool util.DestroyableSessionPool, f func(sctx sessionctx.Context) error, flags ...int) (err error) {
 	defer util.Recover(metrics.LabelStats, "CallWithSCtx", nil, false)
 	se, err := pool.Get()
 	if err != nil {
-		return err
+		return errors.Trace(err)
 	}
 	defer func() {
 		if err == nil { // only recycle when no error
 			pool.Put(se)
+		} else {
+			// Note: Otherwise, the session will be leaked.
```

Comment:
i think we can add a intest.Assert here, if all errors are unexpected

Reply:
I guess we may have some mock errors here.

Reply:
Some test cases are affected by this assertion, so I removed it and kept only this one.
[`ca7c82f` (#59546)](https://github.com/pingcap/tidb/pull/59546/commits/ca7c82f35daa8ac5c69970aa5e6edd8d7ae3a340)

Reply:
We have other 'put back' codes like https://github.com/pingcap/tidb/blob/72a1145688e5c4cab368a75875cb9eb704b3583e/pkg/statistics/handle/bootstrap.go#L342-L346, I think it's better to handle them as well.

Reply:
Updated.